libraries/core_libs/network/src/tarcap/packets_handlers/common/packet_handler.cpp

```diff
@@ -87,10 +109,10 @@ bool PacketHandler::sealAndSend(const dev::p2p::NodeID& node_id, SubprotocolPack
 }
 
 void PacketHandler::disconnect(dev::p2p::NodeID const& node_id, dev::p2p::DisconnectReason reason) {
-  if (auto host = peers_state_->host_.lock(); host) {
+  if (auto host = peers_state_->host_.lock(); !host) {
```

Comment:
should not be "!host", should be "host". cannot call host->disconnect if host is not exist

Reply:
ooh yeah, nice catch. thanks